src/jit/gentree.h

```diff
@@ -4258,6 +4258,8 @@ struct GenTreeDynBlk : public GenTreeBlk
     GenTreeDynBlk(GenTreePtr addr, GenTreePtr dynamicSize)
         : GenTreeBlk(GT_DYN_BLK, TYP_STRUCT, addr, 0), gtDynamicSize(dynamicSize), gtEvalSizeFirst(false)
     {
+        // Conservatively the 'addr' could be null or point into the global heap.
+        gtFlags |= GTF_EXCEPT | GTF_GLOB_REF;
```

Comment:
Seems like it would be very unusual (and a bit dangerous) for us to use a DynBlk to copy into a fixed size LclVar.